src/daterangepicker/daterangepicker.component.ts

```diff
@@ -717,4 +717,28 @@ export class DaterangepickerComponent implements OnInit {
         this.choosedDate.emit({chosenLabel: '', startDate: null, endDate: null});
         this.hide();
     }
+
+    /**
+     * Find out if the selected range should be disabled if it doesn't
+     * fit into minDate and maxDate limitations.
+     */
+    disableRange(range) {
+      if(range === this.locale.customRangeLabel){
+        return false;
+      }
+      const rangeMarkers = this.ranges[range];
+      let disabled = false;
+      const areBothBefore = rangeMarkers.every( date => {
+        return date.isBefore(this.minDate)
+      });
+
+      const areBothAfter = rangeMarkers.every( date => {
+        return date.isAfter(this.maxDate)
+      });
+
+      if(areBothBefore || areBothAfter) {
```

Comment:
could you remove  disabled variable and directly return the test?
````
const areBothBefore = rangeMarkers.every( date => {
        return date.isBefore(this.minDate)
      });

const areBothAfter = rangeMarkers.every( date => {
        return date.isAfter(this.maxDate)
     });
return (areBothBefore || areBothAfter;)

Reply:
@praveenpuglia just this and will merge :)

Reply:
Just committed the changes. review once :)